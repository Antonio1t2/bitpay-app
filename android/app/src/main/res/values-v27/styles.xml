<?xml version="1.0" encoding="utf-8"?>
<resources>

    <style name="PoweredBy.Main">
        <item name="android:statusBarColor">@color/dosh_core_navigation_bar_background</item>
        <item name="android:windowLightStatusBar">true</item>
        <item name="android:windowLightNavigationBar">@color/dosh_core_navigation_bar_background</item>
    </style>
</resources>