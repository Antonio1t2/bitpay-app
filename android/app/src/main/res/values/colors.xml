<resources>
    <color name="bootsplash_background">#1A3B8B</color>
</resources>
