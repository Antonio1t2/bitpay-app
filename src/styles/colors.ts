/* Primary */
export const Midnight = '#0C204E';
export const BitPay = '#1A3B8B';
export const Action = '#2240C4';

/* Secondary */
export const SlateDark = '#434D5A';
export const Slate = '#9BA3AE';
export const Air = '#EBEDF8';
export const Feather = '#F6F7FC';

/* Neutrals */
export const Black = '#303133';
export const Grey = '#E7E7E7';
export const White = '#FFFFFF';

/* Interface */
export const Caution = '#ef476f';
export const Success = '#2fcfa4';
export const Warning = '#fdb455';
