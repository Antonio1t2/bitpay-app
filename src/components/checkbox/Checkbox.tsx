import React from 'react';
import {RoundedCheckbox} from 'react-native-rounded-checkbox';
import Check from '../../../assets/img/check.svg';
import {Action, SlateDark, White} from '../../styles/colors';
import styled from 'styled-components/native';

interface Props {
  onPress: () => any;
  checked: boolean;
  disabled?: boolean;
}

interface BorderProps {
  checked: boolean;
  disabled: boolean | undefined;
}

const Border = styled.View`
  border: 1px solid
    ${({checked, disabled}: BorderProps) => {
      if (disabled) {
        return 'transparent';
      }

      return checked ? Action : SlateDark;
    }};
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  border-radius: 5px;
`;

const baseStyles = {
  borderRadius: 5,
  width: 30,
  height: 30,
};

const Checkbox: React.FC<Props> = ({onPress, checked, disabled}) => {
  return (
    <RoundedCheckbox
      onPress={() => !disabled && onPress()}
      active={checked}
      uncheckedColor={!disabled ? White : undefined}
      checkedColor={Action}
      innerStyle={{
        ...baseStyles,
      }}
      // @ts-ignore --> testing
      testID="checkbox"
      outerStyle={{
        ...baseStyles,
        borderColor: 'transparent',
      }}>
      <Border checked={checked} disabled={disabled} testID="checkboxBorder" />
      <Check />
    </RoundedCheckbox>
  );
};

export default Checkbox;
