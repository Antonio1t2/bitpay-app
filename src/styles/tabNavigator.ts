import {Action, NeutralSlate, SlateDark, White} from './colors';
import {Platform} from 'react-native';
import {MaterialTopTabNavigationOptions} from '@react-navigation/material-top-tabs';

export const ScreenOptions = (
  width: number,
): MaterialTopTabNavigationOptions => {
  const gutter = 8;
  const totalWidth = width * 2 + gutter * 4;

  return {
    swipeEnabled: false,
    tabBarIndicatorStyle: {
      height: 40,
      borderRadius: 50,
      backgroundColor: Action,
      width: width,
      margin: gutter,
    },
    tabBarActiveTintColor: White,
    tabBarInactiveTintColor: SlateDark,
    tabBarPressColor: NeutralSlate,
    tabBarLabelStyle: {
      fontSize: 16,
      textTransform: 'none',
      fontWeight: '500',
      paddingVertical: Platform.select({
        ios: 4,
        android: 2,
      }),
    },
    tabBarStyle: {
      width: totalWidth,
      alignSelf: 'center',
      borderRadius: 50,
      backgroundColor: NeutralSlate,
      elevation: 0,
      height: 56,
    },
  };
};
