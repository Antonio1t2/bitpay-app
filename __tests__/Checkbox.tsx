import React from 'react';
import Checkbox from '../src/components/checkbox/Checkbox';
import {fireEvent, render} from '@testing-library/react-native';
import {SlateDark, Action} from '../src/styles/colors';

it('renders correctly', async () => {
  const mockFn = jest.fn();
  const {toJSON, getByTestId, rerender} = render(
    <Checkbox onPress={mockFn} checked={false} />,
  );
  const checkbox = await getByTestId('checkbox');
  expect(getByTestId('checkboxBorder')).toHaveStyle({borderColor: SlateDark});
  fireEvent(checkbox, 'press');
  expect(mockFn).toHaveBeenCalled();
  expect(toJSON()).toMatchSnapshot();

  rerender(<Checkbox onPress={mockFn} checked={true} />);
  expect(getByTestId('checkboxBorder')).toHaveStyle({borderColor: Action});
  expect(toJSON()).toMatchSnapshot();
});
