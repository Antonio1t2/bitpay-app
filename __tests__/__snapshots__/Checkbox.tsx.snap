// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "transparent",
      "borderRadius": 5,
      "borderWidth": 0,
      "height": 30,
      "justifyContent": "center",
      "transform": Array [
        Object {
          "scale": 0.9,
        },
      ],
      "width": 30,
    }
  }
  testID="checkbox"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 20,
          "height": 40,
          "justifyContent": "center",
          "width": 40,
        },
        Object {
          "borderRadius": 5,
          "height": 30,
          "width": 30,
        },
        Object {
          "backgroundColor": "#FFFFFF",
        },
      ]
    }
  >
    <View
      checked={false}
      style={
        Array [
          Object {
            "borderColor": "#434D5A",
            "borderRadius": 5,
            "borderStyle": "solid",
            "borderWidth": 1,
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
        ]
      }
      testID="checkboxBorder"
    />
    <svg
      data-file-name="SvgCheck"
    />
  </View>
</View>
`;

exports[`renders correctly 2`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "transparent",
      "borderRadius": 5,
      "borderWidth": 1,
      "height": 30,
      "justifyContent": "center",
      "transform": Array [
        Object {
          "scale": 0.9,
        },
      ],
      "width": 30,
    }
  }
  testID="checkbox"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 20,
          "height": 40,
          "justifyContent": "center",
          "width": 40,
        },
        Object {
          "borderRadius": 5,
          "height": 30,
          "width": 30,
        },
        Object {
          "backgroundColor": "#2240C4",
        },
      ]
    }
  >
    <View
      checked={true}
      style={
        Array [
          Object {
            "borderColor": "#2240C4",
            "borderRadius": 5,
            "borderStyle": "solid",
            "borderWidth": 1,
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
        ]
      }
      testID="checkboxBorder"
    />
    <svg
      data-file-name="SvgCheck"
    />
  </View>
</View>
`;
